src/widget/wtracktableview.cpp

```diff
@@ -1129,6 +1122,12 @@ void WTrackTableView::dropEvent(QDropEvent * event) {
                 firstRowToSelect = firstRowToSelect - selectedRowCount;
             }
 
+           if (destIndex.row() < 0) {
+                // If we're moving tracks at invalid position a.k.a. at the very bottom
+                // we have to set first row to reselect correctly
+                firstRowToSelect = model()->rowCount() - selectedRowCount;
```

Comment:
firstRowToSelect?? 
Can we find a better name and improve the comments of the whole section? 
firstRowToResoreSelection
selectionRestoreStartRow